### DISCUSSION TITLE
MySQL collation errors

### DISCUSSION THREAD

**Discussion by 720720** [directus/directus#11786] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

https://github.com/directus/directus/commit/2d0c23d3c2b9d37737a3fe50bbbe1b3a1a7e5f44

<br>

```
my.cnf character_set_server = utf8mb4 collation_server = utf8mb4_general_ci
.env NO DB_CHARSET
```
warning for directus tables utf8mb4_bin

<br>

```
my.cnf character_set_server = utf8mb4 collation_server = utf8mb4_general_ci
.env DB_CHARSET="utf8mb4"
```
warning for directus tables utf8mb4_bin

<br>

```
my.cnf character_set_server = utf8mb4 collation_server = utf8mb4_unicode_ci
.env NO DB_CHARSET
```
warning for directus tables utf8mb4_bin

<br>

```
my.cnf character_set_server = utf8mb4 collation_server = utf8mb4_unicode_ci
.env DB_CHARSET="utf8mb4"
```
warning for directus tables utf8mb4_bin
warning for directus tables utf8mb4_general_ci
warning for collection tables created by directus utf8mb4_general_ci
warning for custom tables != utf8mb4_unicode_ci

<br>

This commit will show warning messages for any directus installation.
Are directus table collation warnings necessary?
Do directus and custom tables need the same collation?
Is there any problem as long as custom tables have the same collation independent of the databases default?

### To Reproduce

npm install directus --production --no-optional
npm install mysql --production
npx directus bootstrap

### Errors Shown

root@debian /s/h/directus# /usr/bin/node /srv/http/directus/node_modules/.bin/directus start
09:24:21 ⚠️  PUBLIC_URL should be a full URL
09:24:21 ⚠️  Some tables and columns do not match your database's default collation (utf8mb4_unicode_ci):
                - Table "directus_users": "utf8mb4_unicode_ci"
                  - Column "tags": "utf8mb4_bin"
                  - Column "auth_data": "utf8mb4_bin"
                - Table "directus_presets": "utf8mb4_unicode_ci"
                  - Column "layout_query": "utf8mb4_bin"
                  - Column "layout_options": "utf8mb4_bin"
                  - Column "filter": "utf8mb4_bin"
                - Table "directus_files": "utf8mb4_unicode_ci"
                  - Column "metadata": "utf8mb4_bin"
                - Table "directus_settings": "utf8mb4_unicode_ci"
                  - Column "storage_asset_presets": "utf8mb4_bin"
                  - Column "basemaps": "utf8mb4_bin"
                  - Column "module_bar": "utf8mb4_bin"
                - Table "directus_revisions": "utf8mb4_unicode_ci"
                  - Column "data": "utf8mb4_bin"
                  - Column "delta": "utf8mb4_bin"
                - Table "directus_webhooks": "utf8mb4_unicode_ci"
                  - Column "headers": "utf8mb4_bin"
                - Table "directus_permissions": "utf8mb4_unicode_ci"
                  - Column "permissions": "utf8mb4_bin"
                  - Column "validation": "utf8mb4_bin"
                  - Column "presets": "utf8mb4_bin"
                - Table "directus_panels": "utf8mb4_unicode_ci"
                  - Column "options": "utf8mb4_bin"
                - Table "directus_fields": "utf8mb4_unicode_ci"
                  - Column "options": "utf8mb4_bin"
                  - Column "display_options": "utf8mb4_bin"
                  - Column "translations": "utf8mb4_bin"
                  - Column "conditions": "utf8mb4_bin"
                - Table "directus_collections": "utf8mb4_unicode_ci"
                  - Column "translations": "utf8mb4_bin"
                  - Column "item_duplication_fields": "utf8mb4_bin"

09:24:21 ✨ Server started at http://localhost:8055

### What version of Directus are you using?

9.5.1

### What version of Node.js are you using?

16.13.2

### What database are you using?

MariaDB 10.5.12

### What browser are you using?

Firefox 97

### What operating system are you using?

Debian 11

### How are you deploying Directus?

Locally

**Comment by azrikahar** [comment]
Moving this to discussion as it's more so a question, and the PR itself and [a response](https://github.com/directus/directus/issues/11628#issuecomment-1039409271) to your previous semi-related ticket does touch upon on it, but perhaps @Oreilles can shed more light on the questions/doubts you may have.

**Comment by rijkvanzanten** [comment]
Directus warns you if your tables aren't in the same collation as the database default. This is because Directus will rely on the database default collation when creating new tables/columns, which can cause conflicts when you try to make relationships between tables/columns created in Directus and the ones in the database.

**Reply by 720720** [reply]
[index.ts](https://github.com/directus/directus/blob/main/api/src/database/index.ts) validateDatabaseCharset generates utf8mb4_bin warnings even if there is no collation mismatch.

**Reply by rijkvanzanten** [reply]
What is the result if you run the following SQL query directly in the database?

```sql
SELECT @@collation_database as collation
```

**Reply by 720720** [reply]
utf8mb4_unicode_ci

**Reply by rijkvanzanten** [reply]
Hence the warnings right? The tables/columns in your database are using `utf8mb4_bin`, which is different from the globally configured default `utf8mb4_unicode_ci`

**Reply by 720720** [reply]
@rijkvanzanten @Oreilles

Every new Directus installation > 9.5.1 will report some of its own tables.

**Reply by rijkvanzanten** [reply]
> Every new Directus installation > 9.5.1 will report some of its own tables.

That totally depends on your current database configuration.. Directus doesn't set anything explicitly, so will fallback to whatever the database itself is configured to. In the validation check, it simply checks if the table collations match the collation of `@@collation_database` as that's what the `mysql` package defaults to as well

https://github.com/directus/directus/blob/d987b06f5895f231387c8f4feee0c64533d1131f/api/src/database/index.ts#L279

**Reply by 720720** [reply]
| @@character_set_database | @@collation_database |
| ------------------------ | -------------------- |
| latin1                   | latin1_swedish_ci    |

On Debian 11 and MariaDB 10.5.12 all Directus LONGTEXT columns have the collation utf8mb4_bin independent of my database charset and collation settings.

**Reply by martin-braun** [reply]
Am I at risk of getting the same non-sense issue when working with PostgreSQL?

**Comment by 720720** [comment]
Debian 11
MariaDB 10.5.12

```
/etc/mysql/my.cnf

[client]
default_character_set = utf8mb4

[mysql]
default_character_set = utf8mb4

[mysqld]
character_set_server = utf8mb4
collation_server = utf8mb4_unicode_ci
```

```
/srv/http/directus/.env

PORT="8055"

DB_CLIENT="mysql"
DB_HOST="localhost"
DB_PORT="3306"
DB_DATABASE="directus"
DB_USER="directus"
DB_PASSWORD="directus"
```

```
mysql -e "CREATE DATABASE directus CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;"

mkdir -p /srv/http/directus/uploads
cd /srv/http/directus

npm install directus --production --no-optional
npm install mysql --production
npx directus bootstrap

/usr/bin/node /srv/http/directus/node_modules/.bin/directus start
```

<br>

https://github.com/directus/directus/commit/2d0c23d3c2b9d37737a3fe50bbbe1b3a1a7e5f44 generates utf8mb4_bin warnings on a clean directus installation

1. the [database api](https://github.com/directus/directus/blob/main/api/src/database/index.ts) should not report tables that [directus bootstrap](https://github.com/directus/directus/blob/main/api/src/cli/commands/bootstrap/index.ts) creates
2. [directus bootstrap](https://github.com/directus/directus/blob/main/api/src/cli/commands/bootstrap/index.ts) should not create tables that the [database api](https://github.com/directus/directus/blob/main/api/src/database/index.ts) reports

**Reply by cpt-n3mo** [reply]
im having the same trouble with the latest docker image and using a fresh mariadb container. 
this happens with fresh project! 
cant make any relations , 
cant use image relations either, cant even remove the image relation once it created.. so this error sort of makes it unusable.. 

```
 00:45:24 ⚠️  Some tables and columns do not match your database's default collation (utf8mb4_general_ci):
 		- Table "directus_collections": "utf8mb4_general_ci"
 		  - Column "translations": "utf8mb4_bin"
 		  - Column "item_duplication_fields": "utf8mb4_bin"
 		- Table "directus_files": "utf8mb4_general_ci"
 		  - Column "metadata": "utf8mb4_bin"
 		- Table "directus_settings": "utf8mb4_general_ci"
 		  - Column "storage_asset_presets": "utf8mb4_bin"
 		  - Column "basemaps": "utf8mb4_bin"
 		  - Column "module_bar": "utf8mb4_bin"
 		- Table "directus_fields": "utf8mb4_general_ci"
 		  - Column "options": "utf8mb4_bin"
 		  - Column "display_options": "utf8mb4_bin"
 		  - Column "translations": "utf8mb4_bin"
 		  - Column "conditions": "utf8mb4_bin"
 		- Table "directus_webhooks": "utf8mb4_general_ci"
 		  - Column "headers": "utf8mb4_bin"
 		- Table "directus_panels": "utf8mb4_general_ci"
 		  - Column "options": "utf8mb4_bin"
 		- Table "directus_users": "utf8mb4_general_ci"
 		  - Column "tags": "utf8mb4_bin"
 		  - Column "auth_data": "utf8mb4_bin"
 		- Table "directus_presets": "utf8mb4_general_ci"
 		  - Column "layout_query": "utf8mb4_bin"
 		  - Column "layout_options": "utf8mb4_bin"
 		  - Column "filter": "utf8mb4_bin"
 		- Table "directus_permissions": "utf8mb4_general_ci"
 		  - Column "permissions": "utf8mb4_bin"
 		  - Column "validation": "utf8mb4_bin"
 		  - Column "presets": "utf8mb4_bin"
 		- Table "directus_revisions": "utf8mb4_general_ci"
 		  - Column "data": "utf8mb4_bin"
 		  - Column "delta": "utf8mb4_bin"


```




anyone have a solution for this..?

**Reply by simonmaris** [reply]
same here

**Reply by nightlyworker** [reply]
Solved with:
`create DATABASE XXXXX CHARACTER SET utf8mb4 COLLATE utf8mb4_bin;`

**Comment by domjag** [comment]
Had the same issue on a fresh install:
Windows 10 Xampp: Maria DB 10.4.13
Created DB with utf8mb4_general_ci - Gave me the same error: **Some tables and columns do not match your database's default collation (utf8mb4_general_ci):** every table pooched.
-removed everything and started over but this time I set charset to utf8mb4_bin - no issues with tables

2nd issue: env variable was set to 0.0.0.0 instead of 127.0.0.1 after fresh install -had to manually correct it. Then everything ran ok.

**Comment by 720720** [comment]
Debian 11
MariaDB 10.5.12

```
/etc/mysql/my.cnf

[client]
default_character_set = latin1

[mysqld]
character_set_server = latin1
collation_server = latin1_swedish_ci
```

```
mysql -e "CREATE DATABASE directus CHARACTER SET latin1 COLLATE latin1_swedish_ci;"

mkdir -p /srv/http/directus/uploads
cd /srv/http/directus

npm install directus --production --no-optional
npm install mysql --production
npx directus bootstrap

/usr/bin/node /srv/http/directus/node_modules/.bin/directus start
```

<br>

15:58:49 ⚠️  Some tables and columns do not match your database's default collation (latin1_swedish_ci):
                - Table "directus_users": "latin1_swedish_ci"
                  - Column "tags": "utf8mb4_bin"
                  - Column "auth_data": "utf8mb4_bin"
                - Table "directus_presets": "latin1_swedish_ci"
                  - Column "layout_query": "utf8mb4_bin"
                  - Column "layout_options": "utf8mb4_bin"
                  - Column "filter": "utf8mb4_bin"
                - Table "directus_files": "latin1_swedish_ci"
                  - Column "metadata": "utf8mb4_bin"
                - Table "directus_settings": "latin1_swedish_ci"
                  - Column "storage_asset_presets": "utf8mb4_bin"
                  - Column "basemaps": "utf8mb4_bin"
                  - Column "module_bar": "utf8mb4_bin"
                - Table "directus_revisions": "latin1_swedish_ci"
                  - Column "data": "utf8mb4_bin"
                  - Column "delta": "utf8mb4_bin"
                - Table "directus_webhooks": "latin1_swedish_ci"
                  - Column "headers": "utf8mb4_bin"
                - Table "directus_permissions": "latin1_swedish_ci"
                  - Column "permissions": "utf8mb4_bin"
                  - Column "validation": "utf8mb4_bin"
                  - Column "presets": "utf8mb4_bin"
                - Table "directus_panels": "latin1_swedish_ci"
                  - Column "options": "utf8mb4_bin"
                - Table "directus_fields": "latin1_swedish_ci"
                  - Column "options": "utf8mb4_bin"
                  - Column "display_options": "utf8mb4_bin"
                  - Column "translations": "utf8mb4_bin"
                  - Column "conditions": "utf8mb4_bin"
                - Table "directus_collections": "latin1_swedish_ci"
                  - Column "translations": "utf8mb4_bin"
                  - Column "item_duplication_fields": "utf8mb4_bin"

<br>

| @@character_set_database | @@collation_database |
| ------------------------ | -------------------- |
| latin1                   | latin1_swedish_ci    |

On Debian 11 and MariaDB 10.5.12 all Directus LONGTEXT columns have the collation utf8mb4_bin independent of my database charset and collation settings.

**Reply by piotr-cz** [reply]
Same with MariaDB 10.4

| @@character_set_database | @@collation_database |
| --- | --- |
| utf8mb4 | utf8mb4_unicode_ci |

**Comment by piotr-cz** [comment]
It seems that after running `npx directus bootstrap` some table columns are created with `utf8mb4_bin` collation.
This affects all columns with Datatype: LONGTEXT

The `utf8mb4_bin` is first of `utf8mb4_*` collations so I'd say that because of a bug first collation in list is picked instead of database collation.

Database: MariaDB 10.4, doesn't happen in MySQL 5.7

**Reply by piotr-cz** [reply]
After short research I see that MariaDB for JSON datatypes uses LONGTEXT..

> JSON is an alias for [LONGTEXT](https://mariadb.com/kb/en/longtext/) introduced for compatibility reasons with MySQL's JSON data type.

(see MariaDB > [JSON Data Type](https://mariadb.com/kb/en/json-data-type/))

..with collation `utf8mb4_bin`
(see DZone > [Using JSON in MariaDB](https://dzone.com/articles/using-json-in-mariadb))

So these warnings should be safe to ignore for MariaDB.
Ideally there would be a switch to hide them using an environment variable.

**Reply by piotr-cz** [reply]
I've created PR for this issue here: #14787

**Reply by piotr-cz** [reply]
I think it's safe to change columns collation manually to database default.
At least I didn't find any side effects yet.

**Reply by derrickmehaffy** [reply]
Since the Directus team don't want to fix the issue, if someone needs to build a quick script to do this, here you go (for non MariaDB users, change the charset and collate as needed.

```sql
ALTER DATABASE directus CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;

ALTER TABLE directus.directus_panels MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_fields MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `display_options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `translations` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `conditions` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `validation` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_users MODIFY COLUMN `tags` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_users MODIFY COLUMN `auth_data` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_revisions MODIFY COLUMN `data` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_revisions MODIFY COLUMN `delta` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_flows MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_settings MODIFY COLUMN `storage_asset_presets` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `basemaps` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `module_bar` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `custom_aspect_ratios` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_webhooks MODIFY COLUMN `headers` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_collections MODIFY COLUMN `translations` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_collections MODIFY COLUMN `item_duplication_fields` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_operations MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_permissions MODIFY COLUMN `permissions` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_permissions MODIFY COLUMN `validation` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_permissions MODIFY COLUMN `presets` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_files MODIFY COLUMN `metadata` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_presets MODIFY COLUMN `layout_query` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_presets MODIFY COLUMN `layout_options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_presets MODIFY COLUMN `filter` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
```

**Reply by fusscreme** [reply]
dont' forget `ALTER DATABASE directus CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;` and then `npx directus database migrate:latest`

**Reply by derrickmehaffy** [reply]
> dont' forget `ALTER DATABASE directus CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;` and then `npx directus database migrate:latest`

Added, thanks

**Reply by krga** [reply]
Thanks @derrickmehaffy & @fusscreme 

If successfully executed the SQL statements & did the DB migration. Here I've found some tables that still were not fixed (directus_settings & directus_users). Here is my full SQL statement which removes the DB warning;

```sql
ALTER DATABASE directus CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;

ALTER TABLE directus.directus_panels MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_fields MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `display_options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `translations` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `conditions` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_fields MODIFY COLUMN `validation` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_users MODIFY COLUMN `tags` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_users MODIFY COLUMN `auth_data` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_revisions MODIFY COLUMN `data` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_revisions MODIFY COLUMN `delta` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_flows MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_settings MODIFY COLUMN `storage_asset_presets` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `basemaps` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `module_bar` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `custom_aspect_ratios` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_webhooks MODIFY COLUMN `headers` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_collections MODIFY COLUMN `translations` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_collections MODIFY COLUMN `item_duplication_fields` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_operations MODIFY COLUMN `options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_permissions MODIFY COLUMN `permissions` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_permissions MODIFY COLUMN `validation` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_permissions MODIFY COLUMN `presets` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_files MODIFY COLUMN `metadata` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_presets MODIFY COLUMN `layout_query` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_presets MODIFY COLUMN `layout_options` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_presets MODIFY COLUMN `filter` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_settings MODIFY COLUMN `theme_light_overrides` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `theme_dark_overrides` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_settings MODIFY COLUMN `public_registration_email_filter` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;

ALTER TABLE directus.directus_users MODIFY COLUMN `theme_light_overrides` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
ALTER TABLE directus.directus_users MODIFY COLUMN `theme_dark_overrides` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci DEFAULT NULL NULL;
```

**Comment by piotr-cz** [comment]
I've created #14787 to fix this issue, which is now closed.
If anyone would like to take it over, please do so.
